### DISCUSSION TITLE
Tinxy 16amp smart switch device and wiring the GPIO04 to High Volt AC line.

### DISCUSSION THREAD

**Discussion by NsinghP** [arendst/Tasmota#18458] [Q&A] [answered]
I have a question on Tinxy Retro Smart Switch. I am planning on flashing Tasmota on this device. (https://templates.blakadder.com/tinxy_1_node_16a.html)  this page says GPIO04 is mapped to Switch 1. The official wiring diagram shows AC220V feeding into the Switch port. I am wondering if AC220v is being used as a signal so the relay gets activated. The point being can I use low amperage switch to signal with 220V AC or Do I need 16amp switch? Am I safe in assuming that the appliance power source would be between Live wire input and Load wire output regardless of which method (MQTT command or physical retro switch) is used to turn on/off the appliance? Or is the switch port being used to power the appliance?
![Screenshot (200)](https://user-images.githubusercontent.com/69470381/233136816-27b6dfb8-9044-45f7-bf49-88bbdd74daa4.png)

![Tinxy16amp](https://user-images.githubusercontent.com/69470381/233136920-ca890a9b-f521-4597-8cf3-79327926a38e.jpeg)

**Comment by barbudor** [comment]
No, the switch input is not used to power the device. It is powered by the L and N inputs
But there's likely an opto-coupler based circuit which as 1 side connected to the switch input and the other side connected internaly to the N so when you feed L inside the switch input the current goes through the opto coupler and return through the N.

So you CAN'T connect that switch input to a low-voltage circuit because that would mean connecting you low-voltage circuit to mains (at least to the N line is you wire that correctly).

This device is not suitable to be triggered by a low voltage

Look for a device that have isolated switch inputs. I think some (if not most) of Shellies do support that.

**Reply by barbudor** [reply, marked as answer]
I may have mis-read your question : the device and the load are powered through L and N inputs
So you can use a low-current switch to provide L signal to the switch input. It's just need a few mA to trigger the opto coupler.
But that is still mains voltage so that should be carefully wired and protected

**Reply by NsinghP** [reply]
Thanks. I have ordered this device and I should have it in next 24 hours. I will verify if there is indeed an optocoupler being used and will use necessary safe wiring to connect a small push-on/push-off button switch. I will post the pictures.

**Reply by NsinghP** [reply]
> I may have mis-read your question : the device and the load are powered through L and N inputs So you can use a low-current switch to provide L signal to the switch input. It's just need a few mA to trigger the opto coupler. But that is still mains voltage so that should be carefully wired and protected

Looks like the switch port goes to a chip that is marked MB10F, which according to google is used for AC to DC bridge full wave rectifier. I guess in this case it serves the same purpose as an optocoupler would. Regardless, the switch port trace on the PCB (Marked D2 in the picture) is very thin along with the neutral trace right next to it, so it is not designed to power the load. L and Load traces are
![Tinxy 16amp back](https://user-images.githubusercontent.com/69470381/233395109-ea6d2490-6cd0-4eb1-ac6d-36c8e64cceef.jpeg)
 very thick and fortified so I think I will be safe using 1-3 amp AC220V push type switch to generate on/off signal for Tasmota on GPIO4.

EDIT: the rectifier is feeding into an EL817 C135 Optocoupler so your hypothesis is right.

**Reply by NsinghP** [reply]
>...just need a few mA to trigger the opto coupler. But that is still mains voltage so that should be carefully wired and protected

After thinking it through, I am going to skip using mains voltage to power the switch port. Instead I am going to solder and extend the wires of the "Button 1" which is connected to GPIO00 and extend it to front panel of the junction box.  This way I can use momentary push button to toggle on/off and easily reset the Tasmota device by pressing for 40 seconds if ever needed.

**Comment by lalo-uy** [comment]
The protection is the R12, 390k, so very little current get to the MCU pin.
Less than 1mA.
But U3 seems to be the optocupler.

**Reply by NsinghP** [reply]
Yes, You are right, U3 is the optocoupler and R12 is sitting between the rectifier D2 and Optocoupler U3. I am guessing that Rectifier D2 is converting AC to DC, then Resistor R12 is reducing the DC voltage to acceptable level before sending to Optocoupler U3. EDIT: Also looks like the Mains Neutral is passing through without any protection to the optocoupler. I guess no protection is needed! unless the rectifier is taking care of that.
